akka-http-tests/src/test/scala/akka/http/scaladsl/server/SizeLimitSpec.scala

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.NotUsed
+
+import scala.collection.immutable
+import akka.actor.ActorSystem
+import akka.http.scaladsl.client.RequestBuilding
+import akka.http.scaladsl.coding.{ Decoder, Gzip }
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpEntity.Chunk
+import akka.http.scaladsl.model.headers.{ HttpEncoding, HttpEncodings, `Content-Encoding` }
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{ Flow, Source }
+import akka.testkit.{ EventFilter, TestKit }
+import akka.util.ByteString
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.{ BeforeAndAfterAll, Matchers, WordSpec }
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{ Millis, Seconds, Span }
+
+class SizeLimitSpec extends WordSpec with Matchers with RequestBuilding with BeforeAndAfterAll with ScalaFutures {
+
+  val maxContentLength = 800
+  // Protect network more than memory:
+  val decodeMaxSize = 1600
+
+  val testConf: Config = ConfigFactory.parseString(s"""
+    akka.loggers = ["akka.testkit.TestEventListener"]
+    akka.loglevel = ERROR
+    akka.stdout-loglevel = ERROR
+    akka.http.parsing.max-content-length = $maxContentLength
+    akka.http.routing.decode-max-size = $decodeMaxSize
+    """)
+  implicit val system = ActorSystem(getClass.getSimpleName, testConf)
+  import system.dispatcher
+  implicit val materializer = ActorMaterializer()
+  val random = new scala.util.Random(42)
+
+  implicit val defaultPatience = PatienceConfig(timeout = Span(2, Seconds), interval = Span(5, Millis))
+
+  "a normal route" should {
+    val route = path("noDirective") {
+      post {
+        entity(as[String]) { _ ⇒
+          complete(HttpEntity(ContentTypes.`text/html(UTF-8)`, "<h1>Say hello to akka-http</h1>"))
+        }
+      }
+    }
+
+    val binding = Http().bindAndHandle(route, "localhost", port = 0).futureValue
+
+    "accept small POST requests" in {
+      Http().singleRequest(Post(s"http:/${binding.localAddress}/noDirective", entityOfSize(maxContentLength)))
+        .futureValue.status shouldEqual StatusCodes.OK
+    }
+
+    "not accept entities bigger than configured with akka.http.parsing.max-content-length" in {
+      // It went from 1 occurrence to 2 after discarding the entity, I think is due to the retrying nature of `handleRejections`
```

Comment:
Comment doesn't apply to anything any more ;)

Reply:
Though, it would be interesting to know how all these occurrence did happen, not sure it really was because of `handleRejections`.